testcontainers/testcontainers-dotnet
# feat: Add network support to the Kafka container

## What does this PR do?

This PR adds the functionality to support network within a Docker container.

## Why is it important?

Network isolation is crucial for testing in an environment that closely mimics production. It ensures that the containers can communicate within a controlled network, similar to how they would in a real-world scenario.

## Related issues

- Closes #736
- Closes #1314 


## How to test this PR

1. Build and start the Kafka and `kcat` containers.
2. Execute the test cases in `KafkaContainerNetworkTest` to verify message production and consumption.
3. Check the logs to ensure the message "Message produced by kcat" is correctly produced and consumed.


## Follow-ups

- Consider adding the same functionality for the redpanda module.
